src/Compatibility/Microsoft.DotNet.ApiCompatibility/DiagnosticBag.cs

```diff
@@ -73,7 +73,8 @@ public void Add(T difference)
         {
             if (!Filter(difference.DiagnosticId, difference.ReferenceId))
             {
-                _differences.Add(difference);
+                if (!_differences.Contains(difference))
```

Comment:
Could you make this a HashSet and implement Equals / GetHashCode? As it stands, this will end up being O(N) for every addition which could add up.

Reply:
For sure. When added the 'Contains' call I thought about doing that but forgot about it. Will add in a sec.